File: android/src/main/java/com/MythiCode/camerakit/FlutterMethodListener.java
package com.MythiCode.camerakit;

import java.util.List;

import io.flutter.plugin.common.MethodChannel;

public interface FlutterMethodListener {

    void onBarcodesRead(List<String> barcodes);

    void onTakePicture(MethodChannel.Result result, String filePath);

    void onTakePictureFailed(MethodChannel.Result result, String errorCode, String errorMessage);
}

File: android/src/main/kotlin/com/MythiCode/camerakit/FlutterMethodListener.kt
package com.MythiCode.camerakit

import io.flutter.plugin.common.MethodChannel

interface FlutterMethodListener {
    fun onBarcodesRead(barcodes: List<String?>?)
    fun onTakePicture(result: MethodChannel.Result?, filePath: String?)
    fun onTakePictureFailed(
        result: MethodChannel.Result?,
        errorCode: String?,
        errorMessage: String?
    )
}